crates/wasi/src/preview2/network.rs

```diff
@@ -1,46 +1,75 @@
 use crate::preview2::bindings::sockets::network::{Ipv4Address, Ipv6Address};
 use crate::preview2::bindings::wasi::sockets::network::ErrorCode;
-use crate::preview2::ip_name_lookup::ResolveAddressStream;
-use crate::preview2::TrappableError;
-use std::net::SocketAddr;
+use crate::preview2::ip_name_lookup::resolve_addresses;
+use crate::preview2::tcp::{DefaultTcpSocket, SystemTcpSocket, TcpSocket};
+use crate::preview2::{BoxSyncFuture, TrappableError};
+use futures::Future;
+use std::io;
+use std::net::{IpAddr, SocketAddr};
 use std::sync::Arc;
 
 /// A network implementation
 pub trait Network: Sync + Send {
     /// Given a name, resolve to a list of IP addresses
-    fn resolve_addresses(&mut self, name: String) -> ResolveAddressStream;
+    ///
+    /// Unicode domain names are automatically converted to ASCII using IDNA encoding.
+    /// If the input is an IP address string, the address is parsed and returned
```

Comment:
For a large part I (mindlessly) copied these docs from the WIT files :) 
That being said, don't think it's _that_ weird to write the documentation for the consumer instead of the implementor. [There is precedent](https://docs.rs/futures/latest/futures/future/trait.Future.html#tymethod.poll)